File: 14-最长公共前缀.cpp
class Solution {
public:
    string helper(vector<string>& strs, int index){  
        string result = "";

        if(!strs.empty() && index < strs[0].size()){
            int j = 1;
            char current = strs[0][index];
            while(j < strs.size() && index < strs[j].size()){
                if(current != strs[j][index])
                    break;
                j++;
            }
            if(j == strs.size()){
                result += current;
                result += helper(strs, index + 1);
            }
        }
        
        return result;
    }

    string longestCommonPrefix(vector<string>& strs) {
        return helper(strs, 0);
    }
};

File: 14-最长公共前缀.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        result = ''
        i = 0
        
        while strs:
            j = 1
            if i == len(strs[0]):
                break
            else:
                current = strs[0][i]
                while j < len(strs):
                    if i == len(strs[j]) or strs[j][i] != current:
                        break
                    j += 1
                if j != len(strs):
                    break
                result += strs[0][i]
                i += 1
        
        return result
